hypothesis-python/src/hypothesis/extra/pytestplugin.py

```diff
@@ -144,6 +146,25 @@ def pytest_runtest_call(item):
                     raise InvalidArgument(message % (name,))
             yield
         else:
+            # Warn about function-scoped fixtures, excluding autouse fixtures because
+            # the advice is probably not actionable and the status quo seems OK...
+            # See https://github.com/HypothesisWorks/hypothesis/issues/377 for detail.
+            argnames = None
+            for fx_defs in item._request._fixturemanager.getfixtureinfo(
+                node=item, func=item.function, cls=None
+            ).name2fixturedefs.values():
+                if argnames is None:
+                    argnames = frozenset(signature(item.function).parameters)
+                for fx in fx_defs:
+                    if fx.scope == "function" and fx.argname in argnames:
+                        note_deprecation(
+                            "%s uses the %r fixture, but function-scoped fixtures "
+                            "should not be used with Hypothesis tests as they will "
+                            "not be reset between test cases!"
```

Comment:
How about “as they will not be reset between examples”?

“test case” is a generic term, “example” is more Hypothesis-specific.